### PR TITLE
feat(vdc): add resource `cloudavenue_vdc_acl`

### PR DESCRIPTION
**Pull request by eremid**
### Description of your changes

If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### CODE REVIEW

File: examples/resources/cloudavenue_acl/resource.tf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+resource "cloudavenue_acl" "example" {
+}
```
Comment (gaetanars) [inline examples/resources/cloudavenue_acl/resource.tf:2]:
```suggestion
```

You can delete this file

File: internal/provider/vdc/acl_resource.go [hunk 2]
```diff
@@ -0,0 +1,380 @@
...
+
+func (r *aclResource) Init(ctx context.Context, rm *aclResourceModel) (diags diag.Diagnostics) {
+	r.vdc, diags = vdc.Init(r.client, rm.VDC)
+	if diags.HasError() {
+		return
+	}
+
+	if diags.HasError() {
+		return
+	}
```
Comment (gaetanars) [inline internal/provider/vdc/acl_resource.go:81]:
```suggestion
	if diags.HasError() {
		return
	}
```

**Review by gaetanars — DISMISSED** [review]

File: internal/provider/vdc/acl_resource.go [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+			ExternalSubject: nil,
+			AccessLevel:     accessLevel,
+		})
+		oModel.AccessLevel = types.StringValue(accessLevel)
+		outputModel = append(outputModel, oModel)
+	}
+	return output, outputModel, nil
+}
+
+func accessControlListToSharedSet(input []*govcdtypes.AccessSetting) ([]acl.SharedWithModel, error) {
```
Comment (azrod) [inline internal/provider/vdc/acl_resource.go:302]: Why not return `diag.Diagnostics` rather than an `error`

**Review by azrod — COMMENTED** [review]
.

File: internal/provider/vdc/acl_resource.go (hunk 3 above)
Comment (eremid) [reply internal/provider/vdc/acl_resource.go:302]: K.I.S.S ?

**Review by azrod — APPROVED** [review]

**Pull request merged**